src/pool/srv_pool.c

```diff
@@ -97,6 +98,15 @@ static int
 find_hdls_to_evict(struct rdb_tx *tx, struct pool_svc *svc, uuid_t **hdl_uuids,
 		   size_t *hdl_uuids_size, int *n_hdl_uuids, char *machine);
 
+size_t
+pool_hdl_size(struct pool_svc *svc)
+{
+	if (svc->ps_global_version == DS_POOL_GLOBAL_VERSION)
```

Comment:
I think there is a problem with the size determination in this function. In the near-future when DS_POOL_GLOBAL_VERSION is incremented from 1 to 2, then this logic will result in pools with ps_global_version=1 returning the smaller size (but it should return the larger size). It seems any pool with ps_global_version > 0 should return the larger pool_hdl size. Do I have that right (I may be confused)?

Reply:
@bfaccini I think even if struct pool_hdl contents and size stays the same, as soon as DS_POOL_GLOBAL_VERSION is incremented higher than 1 (due to any other pool layout change), there will be a problem here. Just trying to be clear (at the risk of being redundant - apologies if that is the case).

@wangshilong what do you think is ps_global_version > 0 a valid comparison for this purpose? Meaning will it be true that ps_global_version == 0 when the code is working with a "2.0 pool/handle"?

Reply:
Yup, i think this better change as something like this:

if (svc->ps_global_version == 0)
        return sizeof(struct pool_hdl_v0)
else
        return sizeof(struct pool_hdl);

DS_POOL_GLOBAL_VERSION will be 2 in release/2.4 because of EC rotation feature

Reply:
@kccain , in my previous update/comment, I was thinking exactly to what @wangshilong has fully described, because it should at least handle all cases where DS_POOL_GLOBAL_VERSION is raised, until pool_hdl size grows again. @kccain , do you agree ?

Reply:
Yes I think this looks right, to test ps_global_version against the value 0.